lib/utils/helpers.coffee

```diff
@@ -70,3 +67,21 @@ exports.getManifest = (image, deviceType) ->
 	.then(JSON.parse)
 	.catch ->
 		resin.models.device.getManifestBySlug(deviceType)
+
+exports.osProgressHandler = (step) ->
+	_ = require('lodash')
+	rindle = require('rindle')
+	visuals = require('resin-cli-visuals')
+
+	step.on('stdout', _.bind(process.stdout.write, process.stdout))
+	step.on('stderr', _.bind(process.stderr.write, process.stderr))
+
+	step.on 'state', (state) ->
+		return if state.operation.command is 'burn'
+		console.log(exports.stateToString(state))
+
+	bar = new visuals.Progress('Writing Device OS')
+
+	step.on('burn', _.bind(bar.update, bar))
```

Comment:
We can of course, I've just copied this verbatim. Had the same  thought, but didn't want to put more changes.